heat/core/logical.py

```diff
@@ -516,10 +517,11 @@ def sanitize_input_type(
     x = sanitize_input_type(x, y)
     y = sanitize_input_type(y, x)
 
-    # if one of the tensors is distributed, unsplit/gather it
-    if x.split is not None and y.split is None:
-        t1 = manipulations.resplit(x, axis=None)
-        return t1, y
+    # if one of the DNDarrays is distributed and the other is not
+    if x.split is not None and y.split is None and y.ndim > 0:
+        t2 = factories.array(y.larray, device=x.device, split=x.split)
+        x, t2 = sanitation.sanitize_distribution(x, t2, target=x)
+        return x, t2
 
```

Comment:
We're missing the same check for y distributed, x non distributed I think. Otherwise we're good to go.